chia/wallet/wallet_node.py

```diff
@@ -661,16 +673,25 @@ async def receive_and_validate(inner_states: List[CoinState], inner_idx_start: i
                 self.log.error(f"Disconnected from peer {peer.peer_node_id} host {peer.peer_host}")
                 return False
             if trusted:
-                try:
-                    self.log.info(f"new coin state received ({idx}-" f"{idx + len(states) - 1}/ {len(items)})")
-                    await self.wallet_state_manager.new_coin_state(states, peer, fork_height)
-                    await self.wallet_state_manager.blockchain.set_finished_sync_up_to(
-                        last_change_height_cs(states[-1]) - 1
-                    )
-                except Exception as e:
-                    tb = traceback.format_exc()
-                    self.log.error(f"Error adding states.. {e} {tb}")
-                    return False
+                async with self.new_state_lock:
```

Comment:
Why do we need new_state lock? 
Also, we might already be in wallet_state_manage.lock, so it seems like a lot of locking

Reply:
I'm changing this to use db lock, we want to avoid anything else writing to the db during this transaction. (calling commit from elsewhere)